functions/utils/getColor.ts

```diff
@@ -63,5 +63,15 @@ function getAverageColor(arrayBuffer: Uint8Array, type?: string) {
   g = Math.floor(g / (pixelCount - transparentPixels))
   b = Math.floor(b / (pixelCount - transparentPixels))
 
+  if (checkDistance) {
+    const distance = Math.sqrt(Math.pow(r - 255, 2) + Math.pow(g - 255, 2) + Math.pow(b - 255, 2))
+
+    if (distance < 50) {
+      r -= 20
+      g -= 20
+      b -= 20
+    }
+  }
```

Comment:
Should this be a while loop to enforce that distance ends up >=50?
Should there be a check that r,g,b > 0?

```suggestion
  if (checkDistance) {
    let distance = Math.sqrt((r - 255)**2, (g - 255)**2, (b - 255)**2)
    while (distance < 50) {
      r = Math.max(r - 20, 0)
      g = Math.max(g - 20, 0)
      b = Math.max(b - 20, 0)
      distance = Math.sqrt((r - 255)**2, (g - 255)**2, (b - 255)**2)
    }
  }
```